crates/core/tedge_mapper/src/c8y/mapper.rs

```diff
@@ -48,28 +50,44 @@ impl TEdgeComponent for CumulocityMapper {
         let (mut runtime, mut mqtt_actor) =
             start_basic_actors(&c8y_mapper_name, &tedge_config).await?;
 
-        let mqtt_config = tedge_config.mqtt_config()?;
         let c8y_mapper_config =
             C8yMapperConfig::from_tedge_config(cfg_dir, &tedge_config, c8y_profile)?;
         if tedge_config.mqtt.bridge.built_in {
-            let custom_topics = c8y_config
+            let smartrest_1_topics = c8y_config
+                .smartrest1
+                .templates
+                .0
+                .iter()
+                .map(|id| Cow::Owned(format!("s/dl/{id}")));
+
+            let smartrest_2_topics = c8y_config
                 .smartrest
                 .templates
                 .0
                 .iter()
                 .map(|id| Cow::Owned(format!("s/dc/{id}")));
 
+            let use_certificate = c8y_config
+                .auth_method
+                .is_certificate(&c8y_config.credentials_path);
             let cloud_topics = [
-                "s/dt",
-                "s/dat",
-                "s/ds",
-                "s/e",
-                "devicecontrol/notifications",
-                "error",
+                ("s/dt", true),
+                ("s/ds", true),
+                ("s/dat", use_certificate),
```

Comment:
Not a request for change but just a remark: I'd have just kept the older array (or a vec) and just added this `s/dat` topic to that vector based on the `is_certificate` value, to keep the logic simpler.